src/H5E.c

```diff
@@ -1532,16 +1532,22 @@ H5Eget_auto2(hid_t estack_id, H5E_auto2_t *func /*out*/, void **client_data /*ou
     H5E_auto_op_t op;                  /* Error stack function */
     herr_t        ret_value = SUCCEED; /* Return value */
 
-    FUNC_ENTER_API(FAIL)
+    /* Don't clear the error stack! :-) */
+    FUNC_ENTER_API_NOCLEAR(FAIL)
```

Comment:
The changes in this file make it so that H5Eget_auto2/H5Eauto_is_v2 do not clear the error stack. H5Eset_auto2 already avoids clearing the error stack, so it seems logical to do the same for H5Eget_auto2. H5Eauto_is_v2 also seems to make sense since we just want to check the type of error stack.

This was noticed by observing that when an H5E_BEGIN/END_TRY block is used, it would trigger a clear of the error stack and lose anything on the stack before the block, even though the purpose of this block is usually to avoid error printing of something expected to fail (which we don't want to see).

Since these blocks don't actually prevent those error messages from being put on the error stack, these changes have the side effect that those failure messages will probably now show up if an error stack eventually gets printed out; they will just get printed out later on. However, I still think that this is the "correct" behavior. The only reason error messages weren't printed before was due to the hidden stack clear from H5Eget_auto2/H5Eauto_is_v2 that was also blowing away other important and potentially unrelated error messages on the stack.

If this is a major problem, we can re-engineer H5E_BEGIN/END_TRY to pop the expected error messages off the stack in the same way that I do in H5VLcallback.c in this PR.